core/bin/zksync_api/src/fee_ticker/validator/watcher.rs

```diff
@@ -80,40 +135,27 @@ impl UniswapTokenWatcher {
     }
 }
 
-#[derive(Serialize, Deserialize, Debug)]
-pub struct GraphqlResponse {
-    pub data: GraphqlTokenResponse,
-}
-
-#[derive(Serialize, Deserialize, Debug)]
-pub struct GraphqlTokenResponse {
-    pub token: Option<TokenResponse>,
-}
-
-#[derive(Serialize, Deserialize, Debug)]
-pub struct TokenResponse {
-    /// Total amount swapped all time in token pair stored in USD, no minimum liquidity threshold.
-    #[serde(rename = "untrackedVolumeUSD")]
-    pub untracked_volume_usd: String,
-}
-
 #[async_trait::async_trait]
-impl TokenWatcher for UniswapTokenWatcher {
+impl TokenWatcher for CoinGeckoTokenWatcher {
     async fn get_token_market_volume(&mut self, token: &Token) -> anyhow::Result<BigDecimal> {
-        match self.get_market_volume(token.address).await {
-            Ok(amount) => {
-                self.update_historical_amount(token.address, amount.clone())
-                    .await;
-                return Ok(amount);
-            }
-            Err(err) => {
-                vlog::error!("Error in api: {:?}", err);
+        if let Ok(coingecko_coin) = self.find_token_by_symbol(&token.symbol).await {
+            match self.get_market_volume(&coingecko_coin.id).await {
```

Comment:
just realised that there is more reliable set of endpoints:
1. "/asset_platforms" - gives list of networks
2. filter by chain id
3. get platform_id
4. "/coins/{platform_id}/contract/{contract_address}" - get the same info as "/coins/{coin_id}", but removes chance of accidental symbol collision

Reply:
done